pkg/hubble/exporter/config_watcher.go

```
@@ -0,0 +1,122 @@
+	watcher := &configWatcher{
+		logger:         logrus.New().WithField("configFilePath", configFilePath),
+		configFilePath: configFilePath,
+		callback:       callback,
+	}
+
+	// initial configuration load
+	watcher.reload()
+
+	// TODO replace ticker reloads with inotify watchers
+	watcher.ticker = time.NewTicker(reloadInterval)
+	watcher.stop = make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-watcher.stop:
+				return
+			case <-watcher.ticker.C:
+				watcher.reload()
+			}
+		}
+	}()
+
+	return watcher
+}
+
+func (c *configWatcher) reload() {
+	c.logger.Debug("Attempting reload")
+	config, hash, err := c.readConfig()
+	if err != nil {
+		DynamicExporterReconfigurations.WithLabelValues("failure").Inc()
+		c.logger.Warnf("failed reading dynamic exporter config")
+	} else {
+		c.callback(context.TODO(), hash, *config)
+	}
+}
+
+// Stop stops watcher.
+func (c *configWatcher) Stop() {
+	c.ticker.Stop()
+	c.stop <- true
+}
+
+func (c *configWatcher) readConfig() (*dynamicExportersConfig, uint64, error) {
+	config := &dynamicExportersConfig{}
+	yamlFile, err := os.ReadFile(c.configFilePath)
+	if err != nil {
+		return nil, 0, fmt.Errorf("cannot read file '%s' %v", c.configFilePath, err)
+	}
+	if err := yaml.Unmarshal(yamlFile, config); err != nil {
+		return nil, 0, fmt.Errorf("cannot parse yaml %v", err)
+	}
+
+	if err := validateConfig(config); err != nil {
+		return nil, 0, fmt.Errorf("invalid yaml config file %v", err)
+	}
+
+	return config, calculateHash(yamlFile), nil
+}
+
+func calculateHash(file []byte) uint64 {
+	sum := md5.Sum(file)
+	return binary.LittleEndian.Uint64(sum[0:16])
+}
+
+func validateConfig(config *dynamicExportersConfig) error {
+	flowlogNames := make(map[string]bool)
+	flowlogPaths := make(map[string]bool)
+
+	for i := range config.FlowLogs {
+		name := config.FlowLogs[i].Name
+		filePath := config.FlowLogs[i].FilePath
+
+		if _, ok := flowlogNames[name]; ok {
+			return fmt.Errorf("duplicated flowlog name %s", name)
```

nit: Do we want to return an error right away or compile a collection of errors if any?